### ISSUE TITLE
nsock_loop() continuesly loops with 0 events pending

### ISSUE THREAD

**Issue by Adamtimtim** [nmap/nmap#1149]
I'm doing a simple scan of the local network with 

> nmap -p 5900 --script vnc-brute 192.168.0.0/24 --script-args passdb=/root/Desktop/pass.txt 

And due to the volume of IP's it starts getting stuck with NSOCK DEBUG, and the ETA just keeps climbing:

> NSOCK DEBUG [17.9320s] nsock_loop(): nsock_loop() started (timeout=50ms). 0 events pending
NSOCK DEBUG [17.9320s] nsock_set_loglevel(): Set log level to FULL DEBUG

Is there anyway to mitigate this so I can complete the scan?

**Comment by dmiller-nmap** [comment]
I'm a bit confused as to what could be causing this. Can you reduce the scan to a smaller set that still produces the error? For example, do a port scan for 5900, then do your `--script vnc-brute` against each of the discovered systems separately. Does one of them cause the error, or do they all finish?

If you can get it down to just one target that causes the error, then further try to reduce the problem:
* Does it happen with `--script-args unpwdb.passlimit=5`
* Does it happen with `--script-args unpwdb.timelimit=5s`

Finally, what is the output of `nmap --version`?

**Comment by Adamtimtim** [comment]
The nmap version I'm running:

> Nmap version 7.60 ( https://nmap.org )
Platform: x86_64-pc-linux-gnu
Compiled with: liblua-5.3.3 openssl-1.1.0g libssh2-1.8.0 libz-1.2.8 libpcre-8.39 nmap-libpcap-1.7.3 nmap-libdnet-1.12 ipv6
Compiled without:
Available nsock engines: epoll poll select

I had to do the IP ranges from 30 at a time to 10 and then to 5 to stop getting the error. I couldn't bring it down to 1 IP that was the issue. I tried the above script-args and I got the same issue.
